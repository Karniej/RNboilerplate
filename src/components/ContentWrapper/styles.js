import { colors } from '../../config'

const styles = {
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.primaryColor,
    paddingTop: 250
  },
  modalContent: {
    flex: 1,
    justifyContent: 'center',
    paddingTop: 20
  },
  modalButtons: {
    paddingVertical: 5,
    paddingHorizontal: 10,
    flexDirection: 'row',
    justifyContent: 'space-between'
  }
}
export default styles
