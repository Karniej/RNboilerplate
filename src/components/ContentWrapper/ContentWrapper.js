import React from 'react'
import { View } from 'react-native'
import PropTypes from 'prop-types'
import styles from './styles'

const { container } = styles

const ContentWrapper = ({ children, additionalStyles }) => {
  return <View
    style={[container, additionalStyles]}
  >{children}
  </View>
}

ContentWrapper.propTypes = {
  additionalStyles: PropTypes.object,
  children: PropTypes.node
}
export default ContentWrapper
