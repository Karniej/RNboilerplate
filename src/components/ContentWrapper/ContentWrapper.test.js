import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ContentWrapper from './ContentWrapper'

configure({ adapter: new Adapter() })

test('Renders correctly, expect onpress to be defined, but not called', () => {
  const defaultProps = {
    children: {}
  }
  const wrapper = shallow(<ContentWrapper
    {...defaultProps}
  />)
  expect(wrapper).toMatchSnapshot()
})
