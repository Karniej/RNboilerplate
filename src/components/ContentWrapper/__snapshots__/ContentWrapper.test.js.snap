// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly, expect onpress to be defined, but not called 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContentWrapper>
    Object {}
  </ContentWrapper>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        Object {},
        <Logo
          additionalStyle={Object {}}
          onPress={[Function]}
        />,
        <Component
          animationType="slide"
          hardwareAccelerated={false}
          onRequestClose={[Function]}
          url={
            Object {
              "uri": "https://selfmadedeveloper.com/",
            }
          }
          visible={false}
        >
          <Component
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
                "paddingTop": 20,
              }
            }
          >
            <Component
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingHorizontal": 10,
                  "paddingVertical": 5,
                }
              }
            >
              <Icon
                allowFontScaling={false}
                name="ios-arrow-back"
                onPress={[Function]}
                size={25}
              />
              <Icon
                allowFontScaling={false}
                name="ios-browsers-outline"
                onPress={[Function]}
                size={25}
              />
            </Component>
            <WebView
              scalesPageToFit={true}
              source={
                Object {
                  "uri": "https://selfmadedeveloper.com/",
                }
              }
            />
          </Component>
        </Component>,
      ],
      "style": Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5FCFF",
          "flex": 1,
          "justifyContent": "center",
          "paddingTop": 250,
        },
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {},
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "additionalStyle": Object {},
          "onPress": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animationType": "slide",
          "children": <Component
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
                "paddingTop": 20,
              }
            }
          >
            <Component
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingHorizontal": 10,
                  "paddingVertical": 5,
                }
              }
            >
              <Icon
                allowFontScaling={false}
                name="ios-arrow-back"
                onPress={[Function]}
                size={25}
              />
              <Icon
                allowFontScaling={false}
                name="ios-browsers-outline"
                onPress={[Function]}
                size={25}
              />
            </Component>
            <WebView
              scalesPageToFit={true}
              source={
                Object {
                  "uri": "https://selfmadedeveloper.com/",
                }
              }
            />
          </Component>,
          "hardwareAccelerated": false,
          "onRequestClose": [Function],
          "url": Object {
            "uri": "https://selfmadedeveloper.com/",
          },
          "visible": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Component
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingHorizontal": 10,
                    "paddingVertical": 5,
                  }
                }
              >
                <Icon
                  allowFontScaling={false}
                  name="ios-arrow-back"
                  onPress={[Function]}
                  size={25}
                />
                <Icon
                  allowFontScaling={false}
                  name="ios-browsers-outline"
                  onPress={[Function]}
                  size={25}
                />
              </Component>,
              <WebView
                scalesPageToFit={true}
                source={
                  Object {
                    "uri": "https://selfmadedeveloper.com/",
                  }
                }
              />,
            ],
            "style": Object {
              "flex": 1,
              "justifyContent": "center",
              "paddingTop": 20,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Icon
                    allowFontScaling={false}
                    name="ios-arrow-back"
                    onPress={[Function]}
                    size={25}
                  />,
                  <Icon
                    allowFontScaling={false}
                    name="ios-browsers-outline"
                    onPress={[Function]}
                    size={25}
                  />,
                ],
                "style": Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingHorizontal": 10,
                  "paddingVertical": 5,
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "allowFontScaling": false,
                    "name": "ios-arrow-back",
                    "onPress": [Function],
                    "size": 25,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "allowFontScaling": false,
                    "name": "ios-browsers-outline",
                    "onPress": [Function],
                    "size": 25,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "scalesPageToFit": true,
                "source": Object {
                  "uri": "https://selfmadedeveloper.com/",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          Object {},
          <Logo
            additionalStyle={Object {}}
            onPress={[Function]}
          />,
          <Component
            animationType="slide"
            hardwareAccelerated={false}
            onRequestClose={[Function]}
            url={
              Object {
                "uri": "https://selfmadedeveloper.com/",
              }
            }
            visible={false}
          >
            <Component
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                  "paddingTop": 20,
                }
              }
            >
              <Component
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingHorizontal": 10,
                    "paddingVertical": 5,
                  }
                }
              >
                <Icon
                  allowFontScaling={false}
                  name="ios-arrow-back"
                  onPress={[Function]}
                  size={25}
                />
                <Icon
                  allowFontScaling={false}
                  name="ios-browsers-outline"
                  onPress={[Function]}
                  size={25}
                />
              </Component>
              <WebView
                scalesPageToFit={true}
                source={
                  Object {
                    "uri": "https://selfmadedeveloper.com/",
                  }
                }
              />
            </Component>
          </Component>,
        ],
        "style": Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5FCFF",
            "flex": 1,
            "justifyContent": "center",
            "paddingTop": 250,
          },
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {},
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "additionalStyle": Object {},
            "onPress": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animationType": "slide",
            "children": <Component
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                  "paddingTop": 20,
                }
              }
            >
              <Component
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingHorizontal": 10,
                    "paddingVertical": 5,
                  }
                }
              >
                <Icon
                  allowFontScaling={false}
                  name="ios-arrow-back"
                  onPress={[Function]}
                  size={25}
                />
                <Icon
                  allowFontScaling={false}
                  name="ios-browsers-outline"
                  onPress={[Function]}
                  size={25}
                />
              </Component>
              <WebView
                scalesPageToFit={true}
                source={
                  Object {
                    "uri": "https://selfmadedeveloper.com/",
                  }
                }
              />
            </Component>,
            "hardwareAccelerated": false,
            "onRequestClose": [Function],
            "url": Object {
              "uri": "https://selfmadedeveloper.com/",
            },
            "visible": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Component
                  style={
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "paddingHorizontal": 10,
                      "paddingVertical": 5,
                    }
                  }
                >
                  <Icon
                    allowFontScaling={false}
                    name="ios-arrow-back"
                    onPress={[Function]}
                    size={25}
                  />
                  <Icon
                    allowFontScaling={false}
                    name="ios-browsers-outline"
                    onPress={[Function]}
                    size={25}
                  />
                </Component>,
                <WebView
                  scalesPageToFit={true}
                  source={
                    Object {
                      "uri": "https://selfmadedeveloper.com/",
                    }
                  }
                />,
              ],
              "style": Object {
                "flex": 1,
                "justifyContent": "center",
                "paddingTop": 20,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Icon
                      allowFontScaling={false}
                      name="ios-arrow-back"
                      onPress={[Function]}
                      size={25}
                    />,
                    <Icon
                      allowFontScaling={false}
                      name="ios-browsers-outline"
                      onPress={[Function]}
                      size={25}
                    />,
                  ],
                  "style": Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingHorizontal": 10,
                    "paddingVertical": 5,
                  },
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "allowFontScaling": false,
                      "name": "ios-arrow-back",
                      "onPress": [Function],
                      "size": 25,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "allowFontScaling": false,
                      "name": "ios-browsers-outline",
                      "onPress": [Function],
                      "size": 25,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "scalesPageToFit": true,
                  "source": Object {
                    "uri": "https://selfmadedeveloper.com/",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
